### ISSUE TITLE
Align in tblr environment from labularray package

### ISSUE THREAD

**Issue by sergiokapone** [texstudio-org/texstudio#2310]
I нашел very progressive and modern analogue of old tabular/tabularx packages - a tabularray package. It is possible to add a cwl for it? Also it is necessary to add alignment behaviour on TXS button "Align Colums" for tblr environment.

```
\documentclass[]{article}
\usepackage{tabularray}

\begin{document}

\begin{tblr}{|l|l|l|l|}
\hline
\SetCell[r=4]{h,1.5cm} Multirow Cell One & Alpha & \SetCell[r=4]{f,1.5cm} Multirow Cell Two & Alpha \\
                                         & Beta                             &                                                                 & Beta \\
                                         & Gamma                       &                                                                 & Gamma \\
                                         & Delta                           &                                                                 & Delta \\
\hline
\end{tblr}

\end{document}
```

**Comment by foreachthing** [comment]
In my TeXStudio, that cwl is already in there (although automatically generated, but it's there!).
You might need to activate it manually:
<img width="655" alt="image" src="https://user-images.githubusercontent.com/10420187/167083812-70850fc9-2522-4e4f-80d4-44aa29c3ce1b.png">

**Comment by sergiokapone** [comment]
Yes, thank you. But the  quastion about TXS button "Align Colums" for tblr environment still left yet.

**Comment by foreachthing** [comment]
> But the question about TXS button "Align Colums" for tblr environment still left yet. Also signs "&" does not highligted inside tblr environment.

Ah, I see. No, this does not work - yet. I don't know where to edit what to make that work.
I guess you could write a custom macro to rename `{tblr}` to `{tabular}`, align and rename back again ;-)

**Comment by sunderme** [comment]
There is no need to activate tabularray by hand as txs scan usepackages.
The cwl for that package is part of txs.
However the interface (column definition) has changed, so it probably will take a while until all things work.

**Comment by sunderme** [comment]
fixed column detection and table manipulation for tblr&co. Advanced column definition does not work.

**Comment by sunderme** [comment]
looks like the code given is invalid.
tabularray, new interface needs `colspec={cclr}`

Nonetheless there seem to be issues in txs.

**Comment by dbitouze** [comment]
I tested with `texstudio-4.8.8beta4-x86_64.AppImage`. Indeed, alignment works well but:

- The `&` special character isn't highlighted as it is with ordinary `tabular` environments.
- Insertion of rows or columns, cutting and pasting columns works only if the colspec is there. They wouldn't work with:
    ```latex
    \begin{tblr}{}
      110 & 12 & 13 & 14 \\
      21 & 220 & 23 & 24 \\
      31 & 32 & 330 & 34 \\
      41 & 42 & 43 & 440 \\
    \end{tblr}
    ```
- When a column is cut, it cannot be pasted before the first one.

**Comment by dbitouze** [comment]
> yes, definition less tabulars are not supported

Too bad! 😉
